src/core/ConfigManager.cpp

```diff
@@ -51,68 +51,31 @@ static inline QString ensureTrailingSlash(const QString & s )
 ConfigManager * ConfigManager::s_instanceOfMe = NULL;
 
 
-ConfigManager::ConfigManager() :
-	m_workingDir(QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation) + "/lmms/"),
-	m_dataDir("data:/"),
-	m_vstDir(m_workingDir + "vst/"),
-	m_sf2Dir(m_workingDir + SF2_PATH),
-	m_gigDir(m_workingDir + GIG_PATH),
-	m_themeDir(defaultThemeDir()),
-	m_lmmsRcFile(QDir::home().absolutePath() +"/.lmmsrc.xml"),
-	m_version(defaultVersion())
+ConfigManager::ConfigManager() : m_version(defaultVersion())
 {
-	// Detect < 1.2.0 working directory as a courtesy
-	if ( QFileInfo( QDir::home().absolutePath() + "/lmms/projects/" ).exists() )
-                m_workingDir = QDir::home().absolutePath() + "/lmms/";
-
-	if (! qgetenv("LMMS_DATA_DIR").isEmpty())
-		QDir::addSearchPath("data", QString::fromLocal8Bit(qgetenv("LMMS_DATA_DIR")));
-
-	// If we're in development (lmms is not installed) let's get the source and
-	// binary directories by reading the CMake Cache
-	QDir appPath = qApp->applicationDirPath();
-	// If in tests, get parent directory
-	if (appPath.dirName() == "tests") {
-		appPath.cdUp();
+	if (QFileInfo::exists(qApp->applicationDirPath() + PORTABLE_MODE_FILE))
+	{
+		initPortableWorkingDir();
 	}
-	QFile cmakeCache(appPath.absoluteFilePath("CMakeCache.txt"));
-	if (cmakeCache.exists()) {
-		cmakeCache.open(QFile::ReadOnly);
-		QTextStream stream(&cmakeCache);
-
-		// Find the lines containing something like lmms_SOURCE_DIR:static=<dir>
-		// and lmms_BINARY_DIR:static=<dir>
-		int done = 0;
-		while(! stream.atEnd())
-		{
-			QString line = stream.readLine();
-
-			if (line.startsWith("lmms_SOURCE_DIR:")) {
-				QString srcDir = line.section('=', -1).trimmed();
-				QDir::addSearchPath("data", srcDir + "/data/");
-				done++;
-			}
-			if (line.startsWith("lmms_BINARY_DIR:")) {
-				m_lmmsRcFile = line.section('=', -1).trimmed() +  QDir::separator() +
-											 ".lmmsrc.xml";
-				done++;
-			}
-			if (done == 2)
-			{
-				break;
-			}
-		}
-
-		cmakeCache.close();
+	else
+	{
+		initInstalledWorkingDir();
 	}
+	m_dataDir = "data:/";
+	m_vstDir = m_workingDir + "vst/";
+	m_sf2Dir = m_workingDir + SF2_PATH;
+	m_gigDir = m_workingDir + GIG_PATH;
+	m_themeDir = defaultThemeDir();
+	if (! qgetenv("LMMS_DATA_DIR").isEmpty())
+		QDir::addSearchPath("data", QString::fromLocal8Bit(qgetenv("LMMS_DATA_DIR")));
```

Comment:
According to the [coding convention](https://github.com/LMMS/lmms/wiki/Coding-Conventions#braces), you should use braces here.

Reply:
I usually always do, but as it was already like this I kept it as it was, was I supposed to change it ?

Reply:
I didn't know that. I'll leave it up to you.